core/stdlib/std.ncl

```diff
@@ -2969,8 +2969,23 @@
         partial_semver_re = m%"^(0|[1-9]\d*)(?:\.(0|[1-9]\d*))?(?:\.(0|[1-9]\d*))?$"%,
         # An exact version constraint. This one is required to have minor and patch versions, and it's allowed to have a prerelease.
         semver_equals_req_re = m%"^=(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?$"%,
+        # The regex for nickel identifiers.
+        ident_re = "^_*[a-zA-Z][_a-zA-Z0-9-']*$",
+      in
+      let
+        semver_req_re = "(%{partial_semver_re})|(%{semver_equals_req_re})",
+        IdentKeys =
```

Comment:
Should we expose `IdentKeys` as well? While it feels like a small self-explanatory contract, exposing it makes it discoverable, re-usable, and queri-able (if we add documentation). Though it also makes it part of the backward-compatibility. So I don't have a strong opinion either way.

Reply:
I've been playing with json schema lately, and it seems pretty common to have a dict with a regex for the key names. What if we exposed, say, `std.record.KeysMatch` which could get used like

```nickel
{
  foo = 1
} | std.record.KeysMatch "^[a-z]+*"
```

(basically, I'm worried that `IdentKeys` is a bit too specific to be worth making public)

Reply:
Yes, I even think CUE provide built-in syntax for that. Nitpick: should we use `FieldMatches` instead to follow Nickel's jargon? Although the issue with field is that it designates both the content and the name. Could use `FieldNameMatches`, but it's longer. That being said, we have `std.record.fields` which return field names, so `FieldMatches` might be reasonable.

Fair for `IdentKeys`, we can keep it local for now.